File: src/main/java/com/carry_laundry/carry_backend/term/presentation/TermController.java
package com.carry_laundry.carry_backend.term.presentation;

import com.carry_laundry.carry_backend.common.presentation.payload.ApiCommonResponse;
import com.carry_laundry.carry_backend.term.application.service.TermService;
import com.carry_laundry.carry_backend.term.domain.entity.TermMeta;
import com.carry_laundry.carry_backend.term.presentation.api.TermSwagger;
import com.carry_laundry.carry_backend.term.presentation.payload.response.TermCodeResponse;
import com.carry_laundry.carry_backend.term.presentation.payload.response.TermCommonResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import reactor.core.publisher.Mono;

@RestController
@RequestMapping("/api/v1/terms")
@RequiredArgsConstructor
public class TermController implements TermSwagger {

    private final TermService termService;

    @GetMapping("/{termCode}")
    public Mono<ApiCommonResponse<TermCommonResponse>> getTerm(@PathVariable String termCode) {
        return termService.getLastTermByCode(termCode)
            .map(ApiCommonResponse::createSuccessResponse);
    }

    @GetMapping
    public Mono<ApiCommonResponse<TermCodeResponse>> getTermCodes() {
        return termService.getTermMetas()
            .map(TermMeta::getCode)
            .collectList()
            .map(TermCodeResponse::new)
            .map(ApiCommonResponse::createSuccessResponse);
    }

}

File: carry-operation/src/main/kotlin/com/carry/operation/adapter/inbound/rest/TermController.kt
package com.carry.operation.adapter.inbound.rest

import com.carry.operation.adapter.inbound.rest.dto.TermResponse
import com.carry.operation.application.port.inbound.TermQueryUseCase
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/v2/terms")
class TermController(
    private val termQueryUseCase: TermQueryUseCase,
) {

    @GetMapping
    fun getActiveTerms(): ResponseEntity<List<TermResponse>> {
        val terms = termQueryUseCase.getActiveTerms()
        return ResponseEntity.ok(terms.map { TermResponse.from(it) })
    }

    @GetMapping("/required")
    fun getRequiredTerms(): ResponseEntity<List<TermResponse>> {
        val terms = termQueryUseCase.getRequiredTerms()
        return ResponseEntity.ok(terms.map { TermResponse.from(it) })
    }
}
